src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Trace.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Types.Enums;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    internal sealed class TraceFunction : BuiltinFunction
+    {
+        public override bool IsSelfContained => false;
+
+        internal const string IgnoreUnsupportedTypesEnumValue = "I";
+
+        public override bool IsAsync => true;
+        
+        public override bool SupportsParamCoercion => true;
+
+        public TraceFunction()
+            : base("Trace", TexlStrings.AboutTrace, FunctionCategories.Behavior, DType.Boolean /* maybe change to void */, 0, 1, 4, DType.String, BuiltInEnums.TraceSeverityEnum.FormulaType._type, DType.EmptyRecord, BuiltInEnums.TraceOptionsEnum.FormulaType._type)
```

Comment:
>BuiltInEnums.TraceOptionsEnum.FormulaType._type [](http://example.com/codeflow?start=215&length=47)

This is updated, while moving definition from PA since now we support typed enums.